suites/message/transfer.go

```diff
@@ -122,11 +122,12 @@ func MessageTest_ValueTransferSimple(t *testing.T, factories state.Factories) {
 			// create a message to transfer funds from `to` to `from` for amount `transferAmnt` and apply it to the state tree
 			// assert the actor balances changed as expected, the receiver balance should not change if transfer fails
 			if tc.code.IsSuccess() {
-				td.AssertBalance(tc.sender, big_spec.Sub(big_spec.Sub(tc.senderBal, tc.transferAmnt), result.Receipt.GasUsed.Big()))
+				td.AssertActor(tc.sender, tc.senderBal, result.Receipt, 1)
 				td.AssertBalance(tc.receiver, tc.transferAmnt)
 			} else {
 				if tc.code == exitcode.SysErrInsufficientFunds {
-					td.AssertBalance(tc.sender, big_spec.Sub(tc.senderBal, result.Receipt.GasUsed.Big()))
+					// TODO: Hackily re-adding the transfer is bad, consider adding a param indicating whether transfer should be subtracted or not
```

Comment:
Isn't that just `code.IsSuccess`? It doesn't matter what the exit code is otherwise, the value should not move.

These changes have definitely confused me.

Reply:
That's...exactly what I need, yeah, thank you so much! (and don't write code late at night).